labman/db/process.py

```diff
@@ -2778,7 +2778,7 @@ def _format_sample_sheet(self, data, sep=','):
         sample_sheet : str
             the sample sheet string
         """
-
+        is_amplicon = self.assay == 'Amplicon'
```

Comment:
With this addition, there would be THREE places in process.py where we were explicitly checking whether the value of self.assay is equal to the string 'Amplicon' (generate_sample_sheet, generate_prep_information, and here).  I think we're just asking for bugs in which, in the future, we change one or two of them but forget to update them all.  I can't take building on this bad practice and want us to stop it here :)

The most minimal way I can think of to address this would be to place the string 'Amplicon' in a constant somewhere, and then have the comparisons (here and elsewhere) made against that constant.  I suppose we'd probably want to do the same for the string 'Metagenomics' that is (for now) the only other acceptable value of the assay attribute.   (The assay attribute is specifically associated with a SequencingProcess, so perhaps the constants could be associated with that.)   This isn't perfect as the assay values come out of the db, so someone could, theoretically, change them there so they mismatched with the code, but at least it minimizes the change points we'd have to touch in such a case.

Reply:
Thanks for your thoughtful comments! I recall that, once I tried to modify the properties of a `SequencingProcess` object in `test_process.py`, and was blocked. So I guess this `assay` attribute is also safe from user modification. FYI.

Reply:
It is good to know the users can't muck with that value!  I hadn't even thought about that nasty possibility; I'm worried about us *coders* deciding to change that string at some point and not remembering to search for all the instances :)